src/test/java/com/google/step/servlets/VoteServletTest.java

```diff
@@ -21,83 +26,122 @@
 @RunWith(JUnit4.class)
 public class VoteServletTest {
 
-  private static final String DEAL_PATH = "/123";
+  private static final String DEAL_PATH = "/" + DEAL_ID_A;
   private static final String DEAL_PATH_INVALID = "/trash";
   private static final String DIR_ONE = "1";
   private static final String DIR_INVALID = "00";
 
   private VoteServlet servlet;
+  private UserService userService;
+  private UserManager userManager;
   private VoteManager voteManager;
+  private HttpServletResponse response;
+  private PrintWriter writer;
 
   @Before
-  public void setUp() {
+  public void setUp() throws IOException {
+    // mock managers
+    userService = mock(UserService.class);
+    userManager = mock(UserManager.class);
     voteManager = mock(VoteManager.class);
-    servlet = new VoteServlet(voteManager);
+    servlet = new VoteServlet(userService, userManager, voteManager);
+
+    // mock HttpServletResponse
+    response = mock(HttpServletResponse.class);
+    StringWriter stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    // behaviour when user is logged in
+    when(userService.isUserLoggedIn()).thenReturn(true);
+    User currentUser = new User(EMAIL_A, "");
+    when(userService.getCurrentUser()).thenReturn(currentUser);
+    when(userManager.readUserByEmail(EMAIL_A)).thenReturn(USER_A);
```

Comment:
I think it might be better to put this in test methods or write this in another method since setup is called before each @Test method but some test method doesn't need.